### PR TITLE
feat(portal): Add styled errors pages (404, 422, 500)

### LINKED ISSUE #2136: Style 404 and 500 error pages

### PR DESCRIPTION
**Pull request by bmanifold**
Closes #2136 

## Screenshots for `404` and `500`

Screenshot 2024-03-20 at 1 16 46 PM

Screenshot 2024-03-20 at 1 16 30 PM

### CODE REVIEW

**Comment by bmanifold** [comment]
As a quick note, I pretty much just followed this doc page: https://hexdocs.pm/phoenix/custom_error_pages.html

@AndrewDryga If there is a better way of doing this just let me know and I'll update the PR.

**Comment by AndrewDryga** [comment]
LGTM. I always dreamed about nice funny error pages like those https://codepen.io/affectionatelettuce/pen/BarrJKJ but we need a designer for that :).

**Review by AndrewDryga — APPROVED** [review]

**Comment by jamilbk** [comment]
> but we need a designer for that :).

I'm sure there's an AI tool that can make us one!

**Comment by bmanifold** [comment]
Yeah, I'm down to make something funnier, I just did these because it was easy.  I'm no animator though, I'd definitely need to find a pre-made animation if we wanted to do something like you linked.

**Comment by jamilbk** [comment]
> _Cool it!_ You're making too many requests

stable-diffusion-turbo

**Review by jamilbk — APPROVED** [review]

File: elixir/apps/web/lib/web/controllers/error_html/500.html.heex [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  </head>
+  <body class="bg-neutral-50">
+    <div class="flex items-center h-screen p-16 bg-gray-50">
+      <div class="container mx-auto flex flex-col items-center">
+        <div class="flex flex-col gap-6 max-w-md text-center">
+          <img src="/images/logo.svg" class="mr-5 h-32" alt="Firezone Logo">
+          <h2 class="font-extrabold text-9xl text-primary-600">
+            <span class="sr-only">Error</span>500
+          </h2>
+          <p class="text-2xl md:text-3xl">Something went wrong.  We'll get it fixed as soon as possible.</p>
```
Comment (jamilbk) [inline elixir/apps/web/lib/web/controllers/error_html/500.html.heex]:
```suggestion
          <p class="text-2xl md:text-3xl">Something went wrong. We've already been notified and will get it fixed as soon as possible.</p>
```

Comment (jamilbk) [reply elixir/apps/web/lib/web/controllers/error_html/500.html.heex]: just to hopefully save a few support tickets

**Comment by AndrewDryga** [comment]
ChatGPT did an image for us: 
DALL·E 2024-03-20 18 31 04 - A humorous scene for a HTTP 500 error page, suitable for a light grey background, featuring a network technician named Andrew  Andrew, depicted in a c

File: elixir/apps/web/lib/web/controllers/error_html/500.html.heex (hunk 1 above)
Comment (bmanifold) [reply elixir/apps/web/lib/web/controllers/error_html/500.html.heex]: 👍 Updated the PR with your suggestion

**Pull request merged**